react/features/welcome/components/BlankPage.native.js

```diff
@@ -0,0 +1,70 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { ActivityIndicator, View } from 'react-native';
+import { connect } from 'react-redux';
+
+import AbstractBlankPage from './AbstractBlankPage';
+import { blankPageStyles as styles } from './styles';
+
+/**
+ * Mobile implementation of a blank page.
+ *
+ * Since this is the component which is rendered when there is no welcome page,
+ * it will show a progress indicator when there are ongoing network requests
+ * (notably, config.js needs to be loaded before we join a conference).
+ *
+ * The use case which prompted the introduction of this component is mobile
+ * where SDK users probably disable the Welcome page.
+ */
+class BlankPage extends AbstractBlankPage {
```

Comment:
During the development and, especially, in the comments we were kind of saying that we know that BlankPage isn't used on Web right now but it may theoretically become useful in the future and besides it wasn't much of an additional effort anyway. Of course, when it needed to import react-native we had to make a split and you added BlankPage.web.js, BlankPage.native.js, and AbstractBlankPage.js. Which is fine but I would've approached it differently. Here's my train of thought:

I'm introducing a network-activity feature which is going to intercept network requests and provide a React Component called something like NetworkActivityIndicator. There're multiple ways to deal with it outside of the feature. I could go like: I'm going to implement NetworkActivityIndicator on react-native and I may have to add a  corresponding empty file on Web or one with `exports default undefined` and then in BlankPage I'll render the NetworkActivityIndicator if it's defined i.e. implemented. So I would've kept most of the work, redux state and logic inside the feature network-activity and BlankPage wouldn't have cared to figure out how to figure out that there's network activity.

Anyway, I'm not saying your approach is wrong and mine is right - I merely felt like sharing.

Reply:
👍